Scripts/RemediationScripts/Remediate-CdnFrontDoorMinTLSVersion.ps1

```diff
@@ -0,0 +1,1094 @@
+                $resourceName = $_.Name
+                $frontDoorEp = [FrontendEndPointsClass]:: new()
+                try
+                {
+                    Write-Host "Fetching Front Door Endpoint configuration: Resource Group Name: [$($resourceGroupName)], Resource Name: [$($resourceName)]..." -ForegroundColor $([Constants]::MessageType.Info)
+                    Write-Host $([Constants]::SingleDashLine)
+                
+                    $FrontDoorEndpoints = Get-AzFrontDoorCdnCustomDomain -ProfileName $resourceName -ResourceGroupName $resourceGroupName
+                
+                    if($FrontDoorEndpoints)
+                    {
+                        Write-Host "Front Door Configurations successfully fetched." -ForegroundColor $([Constants]::MessageType.Update)
+                        Write-Host $([Constants]::SingleDashLine)
+                    }
+
+                    if($null -ne $FrontDoorEndpoints)
+                    {
+                        foreach ($item in $FrontDoorEndpoints) 
+                        {
+                            $domainName = $item.Name
+                            $domainResourceGroupName = $item.ResourceGroupName
+                            $TypeOfCertificate= @()
+                            $MinTLSVersion = @()
+                    
+                            if($null -ne $AccessMgtToken)
+                            {
+                                $header = "Bearer " + $AccessMgtToken
+                                $headers = @{"Authorization"=$header;"Content-Type"="application/json"; "x-ms-version" ="2013-08-01"}
+                                $uri = [string]:: Format("{0}/subscriptions/{1}/resourceGroups/{2}/providers/Microsoft.Cdn/profiles/{3}/customDomains/{4}?api-version=2021-06-01",$ResourceAppIdURI,$SubscriptionId,$domainResourceGroupName,$resourceName,$domainName)
+                                $customDomainResponse = Invoke-WebRequest -Method GET -Uri $uri -Headers $headers -UseBasicParsing
+                    
+                                if($customDomainResponse.StatusCode -ge 200 -and $customDomainResponse.StatusCode -le 399)
+                                {
+                                    if($null -ne $customDomainResponse.Content)
+                                    {
+                                        $custom = $customDomainResponse.Content | ConvertFrom-Json  
+                                        if($null -ne $custom)
+                                        {
+                                            $TypeOfCertificate = $custom.properties.tlsSettings.certificateType
+                                            $MinTLSVersion = $custom.properties.tlsSettings.minimumTlsVersion
+                                        }
+                                    }
+                                }        
+                            }
+                            if($null -ne $MinTLSVersion)
```

Comment:
We have declared $MinTLSVersion as array, this condition will always be true

Reply:
Even it is declared as collection of objects it is not true always. but still I have used it as a single object now.